x-pack/plugins/stack_connectors/public/connector_types/email/email_params.tsx

```diff
@@ -18,6 +18,22 @@ import { EmailActionParams } from '../types';
 
 const noop = () => {};
 
+export const getFormattedEmailOptions = (
+  searchValue: string,
+  previousOptions: Array<{ label: string }>
+): Array<{ label: string }> => {
+  if (!searchValue.trim()) return previousOptions;
+  const previousEmails: string[] = previousOptions.map((option) => option.label);
+  const newEmails: string[] = searchValue.split(',').reduce((result, email) => {
+    const trimmedEmail: string = email.trim();
+    if (trimmedEmail) result.push(trimmedEmail);
+    return result;
+  }, [] as string[]);
+  const allUniqueEmails: string[] = Array.from(new Set([...previousEmails, ...newEmails]));
```

Comment:
what about using a Set from the beginning? Initialising it to `previousEmails` and then inserting there directly will avoid you having to do this extra step to remove duplicates